pydantic/generics.py

```diff
@@ -0,0 +1,67 @@
+from typing import Any, Dict, Tuple, Type, Union, get_type_hints, no_type_check
+
+from pydantic import BaseModel, create_model
+from pydantic.class_validators import gather_validators
+
+
+class BaseGenericModel:
+    """
+    This might benefit from being modeled more closely on something like typing_extensions.Protocol
```

Comment:
This is more a comment than a docstring, maybe you could move this to a comment or into the PR and add a docstring.